services/opm/opm-version.service.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const { addv } = require('../text-formatters')
+const { BaseService, NotFound, InvalidResponse } = require('..')
+
+module.exports = class OpmVersion extends BaseService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'opm/v',
+      pattern: ':user/:moduleName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'OPM',
+        namedParams: {
+          user: 'openresty',
+          moduleName: 'lua-resty-lrucache',
+        },
+        staticPreview: this.render({ version: 'v0.08' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'opm',
+    }
+  }
+
+  static render({ version }) {
+    const color = '#599059'
+
+    return { message: addv(version), color }
+  }
+
+  async fetch({ user, moduleName }) {
+    const { res } = await this._request({
+      url: `https://opm.openresty.org/api/pkg/fetch`,
+      options: {
+        method: 'HEAD',
+        qs: {
+          account: user,
+          name: moduleName,
+        },
+      },
+      errorMessages: {
+        404: 'module not found',
+      },
```

Comment:
So I take it there's no available JSON api we could integrate with?

Reply:
Yes for now there's no JSON API available.